### PR TITLE
Obsolete useManagedSha1 and ProduceLegacyHmacValues

### LINKED ISSUE #53875: Obsolete HMACSHA1 constructor with `useManagedSha1`
## Background and Motivation

`HMACSHA1` has a constructor that the other HMACSHA2/MD5 classes don't, which is to accept a boolean indicating if `SHA1Managed` should be used as the underlying hash. As of .NET Core, the HMAC implementations are not managed themselves anymore, but instead defer to the platform. As such the `useManagedSha1` parameter is ignored.

## Proposed API

```diff
namespace System.Security.Cryptography {
    public class HMACSHA1 : HMAC {
+       [Obsolete("HMACSHA1 always uses the algorithm implementation provided by the platform. Use the constructor without the useManagedSha1 parameter.")]
        [EditorBrowsable(EditorBrowsableState.Never)]
        public HMACSHA1(byte[] key, bool useManagedSha1);
    }

    public class HMACSHA384 : HMAC {
+       [Obsolete("Producing legacy HMAC values is no longer supported.")]
        public bool ProduceLegacyHmacValues { get; set; }
    }

    public class HMACSHA512 : HMAC {
+       [Obsolete("Producing legacy HMAC values is no longer supported.")]
        public bool ProduceLegacyHmacValues { get; set; }
    }
}
```

An analyzer could be made to update call sites to change simple invocations of the constructor like `new HMACSHA1(key_expression, false)` and `new HMACSHA1(key_expression, true)` to `new HMACSHA1(key_expression)`.

### PR DESCRIPTION
**Pull request by vcsjones**
Closes #53875

### CODE REVIEW

File: src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs [hunk 1]
```diff
@@ -483,6 +483,7 @@ public partial class HMACSHA1 : System.Security.Cryptography.HMAC
         public HMACSHA1() { }
         public HMACSHA1(byte[] key) { }
         [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        [System.ObsoleteAttribute("HMACSHA1 always uses the algorithm implementation provided by the platform. Use the constructor without the useManagedSha1 parameter.", DiagnosticId = "SYSLIB0030", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]
```
Comment (stephentoub) [inline src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs:486]: "Use the constructor" makes it sound like there's only one.  Should it be "use _a_ constructor"?

Comment (vcsjones) [reply src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs:486]: Well, there is only one other one that accepts the `key` parameter which I think is where my mind was going with that. But yeah, there's no point in the language limiting it like there is only one (maybe another will get added later).

File: src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs [hunk 2]
```diff
@@ -537,6 +539,7 @@ public partial class HMACSHA512 : System.Security.Cryptography.HMAC
         public HMACSHA512() { }
         public HMACSHA512(byte[] key) { }
         public override byte[] Key { get { throw null; } set { } }
+        [System.ObsoleteAttribute("ProduceLegacyHmacValues is obsolete. Producing legacy HMAC values is no longer supported.", DiagnosticId = "SYSLIB0029", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]
```
Comment (GrabYourPitchforks) [inline src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs:542]: Huh, surprised we didn't EB-never this.

**Comment by vcsjones** [comment]
@bartonjs polite ping just to make sure you saw this for 6.0. 😊

**Review by bartonjs — APPROVED** [review]

**Comment by bartonjs** [comment]
I'd just seen the issue when looking at api-approved and 6.0.0, but missed the PR.  Thanks :smile:.

**Pull request merged**